pymatviz/heatmap.py

```diff
@@ -0,0 +1,109 @@
+"""Plotter for heatmaps."""
+
+
+from __future__ import annotations
+
+from typing import Any
+
+import matplotlib
+import matplotlib.pyplot as plt
+import pandas as pd
+from matplotlib.colors import Colormap
+
+from pymatviz.colormaps import blend_two
+
+
+def heatmap(
+    df: pd.DataFrame,
```

Comment:
I didn't use a more general data type here (ArrayLike for example) because I think we might need to separate data processing from plotting, to simplify plotters (instead of accepting a whole lot of possible types from the **plotter** side). If you agree with this, I might proceed and implement this change, what do you think?

Meanwhile it think it might be better to group plotters into a separate module (dir)?

Reply:
I think `pd.DataFrame` is plenty general enough. And as you said, `pymatviz` is not about getting the data into the right format for you, just getting the right data format on a canvas.